tests/testGradingMethods.js

```diff
@@ -0,0 +1,346 @@
+const { assert } = require('chai');
+const cheerio = require('cheerio');
+const fs = require('fs-extra');
+const path = require('path');
+
+const config = require('../lib/config');
+const fetch = require('node-fetch');
+const helperServer = require('./helperServer');
+const sqlLoader = require('../prairielib/lib/sql-loader');
+const sqlDb = require('../prairielib/lib/sql-db');
+const sql = sqlLoader.loadSqlEquiv(__filename);
+const io = require('socket.io-client');
+const { setUser, parseInstanceQuestionId, saveOrGrade } = require('./helperClient');
+
+const siteUrl = 'http://localhost:' + config.serverPort;
+const baseUrl = siteUrl + '/pl';
+const defaultUser = {
+  authUid: config.authUid,
+  authName: config.authName,
+  authUin: config.authUin,
+};
+
+const fibonacciSolution = fs.readFileSync(
+  path.resolve(
+    __dirname,
+    '..',
+    'testCourse',
+    'questions',
+    'externalGrade',
+    'codeUpload',
+    'tests',
+    'ans.py'
+  )
+);
+
+let socket = null;
+
+const mockStudents = [
+  { authUid: 'student1', authName: 'Student User 1', authUin: '00000001' },
+  { authUid: 'student2', authName: 'Student User 2', authUin: '00000002' },
+  { authUid: 'student3', authName: 'Student User 3', authUin: '00000003' },
+  { authUid: 'student4', authName: 'Student User 4', authUin: '00000004' },
+];
+
+const waitForExternalGrader = async ($questionsPage, questionsPage) => {
+  return new Promise((resolve, reject) => {
+    try {
```

Comment:
Nit: a Promise executor runs in an implicit try/catch, so we can get rid of this explicit one.

Reply:
Resolved in addd8cc